docs: update ROADMAP.md to include roadmap change process

Alex130469 opened the pull request:
Added a change process for the roadmap as it is a suggested item towards incubation.

Alex130469 committed:
docs: update ROADMAP.md to include roadmap change process

Signed-off-by: Alexander Best <alex.best@datacore.com>

avishnu commented on ROADMAP.md:
```diff
@@ -1,3 +1,62 @@
+
+- Proposal is **open for discussion for at least 5 business days**.
+- Discussion happens via:
+  - **GitHub Issue comments** (preferred)
```
Under Proposal Submission, it says issue or PR, but here it's only issue?

tiagolobocastro commented on ROADMAP.md:
```diff
@@ -1,3 +1,62 @@
+
+## Roadmap change process
+
+For a CNCF sandbox project with a **small maintainer team (≤6 maintainers)**, the roadmap change process should be **lightweight, pragmatic, and efficient**, while still being transparent and community-driven. The process below balances inclusiveness with agility.
```
Do we need to be prescriptive here with the explicitly <=6 ?

tiagolobocastro commented on ROADMAP.md:
```diff
@@ -1,3 +1,62 @@
+
+- If a proposal is contested:
+  - Maintain a respectful discussion in line with the Code of Conduct.
+  - If needed, bring in an external CNCF mentor for mediation.
```
This is probably where the steering committee would intervene?

tiagolobocastro commented on ROADMAP.md:
```diff
@@ -1,3 +1,62 @@
+
+#### 2.1 Proposal Submission
+
+- **Where?** Submit a GitHub Issue or a PR modifying docs/roadmap.md.
```
What is the relation between the roadmap.md and the github project items?
I guess what I'm asking is, is the roadmap only "vision" or is it process?
Let's today we want to merge a bugfix or even a new feature, which was not in the roadmap. Does it mean we have to update the roadmap?

avishnu replied:
Roadmap is a vision for the short and long terms. We don't need to update the roadmap for bug fixes.

tiagolobocastro reviewed (commented):
What is the

avishnu commented on ROADMAP.md:
```diff
@@ -1,3 +1,63 @@
+
+### 4. Continuous Process Review
+
+- Every **6 months**, maintainers **review and adjust** the roadmap change process based on project needs.
```
Should this be
"Maintainers periodically review and ...." instead of Every 6 months?

avishnu replied:
Is this suggestion discarded? It's fine, but just confirming as the conversation is marked as resolved, but the change hasn't gone in.

Alex130469 committed:
docs: update ROADMAP.md

Removed the explicit maintainer count.
Added PR to the discussion methods.
Changed roadmap review cadence.

Signed-off-by: Alexander Best <alex.best@datacore.com>
